scripts/Install

```diff
@@ -0,0 +1,196 @@
+    libssl-dev openssl \
+    libacl1-dev libacl1 \
+    build-essential
+    sudo apt-get install -y libfuse-dev fuse pkg-config
+
+  elif [[ $OS == 'arch' ]]; then
+    sudo pacman -S python3 python3-dev python3-pip python-virtualenv \
+    libssl-dev openssl \
+    libacl1-dev libacl1 \
+    build-essential libfuse-dev \
+    fuse pkg-config
+
+  elif [[ $OS == 'fedora' ]]; then
+    sudo dnf install python3 python3-devel python3-pip python3-virtualenv
+    sudo dnf install openssl-devel openssl
+    sudo dnf install libacl-devel libacl
+    sudo dnf install gcc gcc-c++
+    sudo dnf install redhat-rpm-config
+    sudo dnf install fuse-devel fuse pkgconfig
+
+  elif [[ $OSTYPE == 'darwin' ]]; then
+    brew install python3 openssl
+    brew install pkg-config
+    pip3 install virtualenv
+    brew cask install osxfuse
+
+  elif [[ "$OSTYPE" == "freebsd"* ]]; then
+    pkg install -y python3 openssl fusefs-libs pkgconf
+    pkg install -y git
+    python3.4 -m ensurepip # to install pip for Python3
+    To use the mount command:
+    echo 'fuse_load="YES"' >> /boot/loader.conf
+    echo 'vfs.usermount=1' >> /etc/sysctl.conf
+    kldload fuse
+    sysctl vfs.usermount=1
+
+  else; then
+    echo \n "We can't install the dependencies on this system"
+    echo "Try manually" \n
+
+  fi
+}
+
+install_dependencies(){
+  if [[ $OS == 'debian' || $OS == 'ubuntu' ]]; then
+    sudo apt-get install -y wget gpg
+
+  elif [[ $OS == 'arch' ]]; then
+    sudo pacman -S wget gpg
+
+  elif [[ $OS == 'fedora' ]]; then
+    sudo dnf install wget gpg
+
+  elif [[ $OSTYPE == 'darwin' ]]; then
+    brew install wget gpg
+
+  elif [[ "$OSTYPE" == "freebsd"* ]]; then
+    pkg install -y wget gpg
+
+  else; then
+    echo \n "We can't install the dependencies on this system"
+    echo "Try manually: wget gpg" \n
+
+  fi
+}
+
+upgrade_os(){
+  if [[ $OS == 'debian' || $OS == 'ubuntu' ]]; then
+    sudo apt-get update  > /dev/null
+    sudo apt-get upgrade -y  > /dev/null
+
+  elif [[ $OS == 'arch' ]]; then
+    sudo pacman -Su  > /dev/null
+
+  elif [[ $OS == 'fedora' ]]; then
+    sudo yum update > /dev/null
+    sudo yum upgrade -y > /dev/null
+
+  elif [[ "$OSTYPE" == "freebsd"* ]]; then
+    freebsd-update fetch
+    freebsd-update install
+  fi
+}
+
+from_source(){
+  upgrade_os
+
+  install_all_dependencies
+
+  # get borg from github
+  git clone https://github.com/borgbackup/borg.git
+
+  virtualenv --python=python3 borg-env
+  source borg-env/bin/activate   # always before using!
+
+  # install borg + dependencies into virtualenv
+  cd borg
+  pip install -r requirements.d/development.txt
+  pip install -r requirements.d/docs.txt  # optional, to build the docs
+  pip install -r requirements.d/fuse.txt  # optional, for FUSE support
+  pip install -e .  # in-place editable mode
+
+  # optional: run all the tests, on all supported Python versions
+  # requires fakeroot, available through your package manager
+  fakeroot -u tox
```

Comment:
nope.

this is to run the tests. we do this while developing borg and for release testing, but it is not intended to be part of a install-borg script.